src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -432,16 +437,41 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
       placement.inlineSize,
     ]);
 
+    /**
+     * Returns true if the AppLayout is nested
+     * Does not apply to iframe
+     */
+    const getIsNestedInAppLayout = (element: HTMLElement | null, rootClassName: string): boolean => {
+      let currentElement: Element | null = element?.parentElement ?? null;
+
+      while (currentElement) {
+        if (currentElement.classList.contains(rootClassName)) {
```

Comment:
> There is no guarantees rootClassName will remain unchanged across versions

In this case the test I introduce will fail. Isn't it enough to avoid such a scenario?

Reply:
We cannot get multiple app layouts of different versions in our integration testing.

Also, when the test will fail, it will be too late, because we want to ship forward compatible code now, instead of doing weird backward-compatibility tricks in the future

Reply:
> We cannot get multiple app layouts of different versions in our integration testing.

True, but if anyone changes this classname, the integration test will fail anyway.

> getComputedStyle(rootRef.current)[globalVars.stickyVerticalTopOffset] 

I would not say it's more reliable way as this global var might be (in theory) changed / removed / whatever.

So I would keep this condition as simple as possible.

Reply:
I still do not get how this test "should keep header visible and in position while scrolling" will catch multi version environment.

> this global var might be (in theory) changed / removed / whatever

No it won't. It is a part of our stable API (this is why it is called a "global var" and has no hash suffix, unlike other things)

Reply:
Generated class name includes a hash sum of the content. So if any of these styles change (which is plausible), you will get a new class and this code will break: https://github.com/cloudscape-design/components/blob/dc7000bf1f341c3da62ac9228d7ef86fdf587ddb/src/app-layout/visual-refresh-toolbar/skeleton/styles.scss#L30-L46

Reply:
Oh, makes sense now. Thanks for the explanation